engine/class_modules/paladin/sc_paladin_protection.cpp

```diff
@@ -784,6 +818,11 @@ void paladin_t::create_prot_actions()
     active.judgment = new judgment_prot_t( this );
 
   active.divine_resonance = new avengers_shield_dr_t( this );
+
+  if ( specialization() == PALADIN_PROTECTION )
```

Comment:
I would replace the spec check with
`if (sets->has_set_bonus( PALADIN_PROTECTION, T28, B4 ) )`